ffi/src/lib.rs

```diff
@@ -659,6 +659,41 @@ pub unsafe extern "C" fn get_partition_columns(
     iter.into()
 }
 
+/// Get the domainMetadata for a specific domain in this snapshot
+///
+/// # Safety
+///
+/// Caller is responsible for passing in a valid handle
```

Comment:
can we add some more specifics about this API? it's handing back a string right?  or rather just whatever is done by the allocate_fn?
```suggestion
/// Get the domainMetadata (as a string) for a specific domain in this snapshot.
///
/// # Safety
///
/// Caller is responsible for passing in a valid handle
```